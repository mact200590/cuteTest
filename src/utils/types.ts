import Star from '../components/svg/Star';
export const icons = {
  start: Star,
};

export type IconNames = keyof typeof icons;
