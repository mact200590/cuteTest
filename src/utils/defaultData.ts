// import {Definitions} from '../types';

export const cardDefaultData = {
  author: 'Julio Pérez',
  title:
    'Ya todos los estudiantes cubanos se reincorporaron presencialmente a las aulas - Radio Habana Cuba',
  description:
    'Este lunes más de 718 400 alumnos de preescolar a quinto grados se incorporarán a las aulas, así como los de las provincias de Mayabeque, Villa Clara, Camagüey, Las Tunas, Holguín y Granma, donde no fue posible hacerlo el día 8.',
  url: 'https://www.radiohc.cu/noticias/nacionales/277139-ya-todos-los-estudiantes-cubanos-se-reincorporaron-presencialmente-a-las-aulas',
  urlToImage: 'https://www.radiohc.cu/articles/28751491-pioneroshoy.jpg',
  publishedAt: '2021-11-15T09:06:11Z',
  content:
    'La Habana, 15 nov (RHC) Este lunes más de 718 400 alumnos de preescolar a quinto grados se incorporarán a las aulas, así como los de las provincias de Mayabeque, Villa Clara, Camagüey, Las Tunas, Hol… [+1149 chars]',
} as Definitions.Article;

export const articles = [];

export const momentData = [
  {
    author: 'null',
    title:
      'Cuba retira acreditación a cinco periodistas de la agencia de noticias española EFE - euronews',
    description: 'MADRID, 14 nov  –     El gobierno',
    url: 'https://es.euronews.com/2021/11/14/cuba-protestas-periodistas',
    urlToImage:
      'https://static.euronews.com/articles/wires/795/64/7956480/1000x563_xcc05.jpg',
    publishedAt: '2021-11-14T11:28:29Z',
    content:
      'MADRID, 14 nov – El gobierno cubano retiró la acreditación de prensa a cinco periodistas de la agencia española de noticias EFE, en vísperas de una marcha de protesta liderada por grupos de la oposic… [+1342 chars]',
  },
  {
    author: 'null',
    title:
      'Lanzarán a fines de mes el primer sistema de protección planetaria de la Tierra - CubaSÍ',
    description:
      'El primer sistema de defensa planetario DART será lanzado por la NASA y SpaceX el martes 23 de noviembre. Su finalidad será lanzar un cohete no tripulado a gran velocidad para impactar contra un asteroide y así modificar su trayectoria. No obstante, esa roca …',
    url: 'http://www.cubasi.cu/es/noticia/lanzaran-fines-de-mes-el-primer-sistema-de-proteccion-planetaria-de-la-tierra',
    urlToImage:
      'http://www.cubasi.cu/sites/default/files/gettyimages-53194763.jpg',
    publishedAt: '2021-11-14T11:24:59Z',
    content:
      'El primer sistema de defensa planetario DART será lanzado por la NASA y SpaceX el martes 23 de noviembre.Su finalidad será lanzar un cohete no tripulado a gran velocidad para impactar contra un as… [+1784 chars]',
  },
  {
    author: 'null',
    title: 'Sequía golpea duro a agricultores en Iraq - Prensa Latina',
    description:
      'Bagdad, 14 nov (Prensa Latina) La Organización Internacional para las Migraciones (OIM) informó hoy que en Iraq 447 familias expulsadas por el autodenominado Estado Islámico y regresaron después, abandonaron sus tierras ante una severa sequía.',
    url: 'https://www.prensa-latina.cu/2021/11/14/sequia-golpea-duro-a-agricultores-en-iraq',
    urlToImage:
      'https://www.prensa-latina.cu/wp-content/uploads/2021/11/Iraq-sequia-agricultura.jpg',
    publishedAt: '2021-11-14T08:09:32Z',
    content: 'null',
  },
  {
    author: 'https://www.facebook.com/bbcnews',
    title:
      'COP26: los avances (y las sombras) que deja el nuevo acuerdo mundial por el cambio climático - BBC Mundo',
    description:
      'Por muy defectuoso que sea, el acuerdo de Glasgow de la COP26 mantiene vivas las esperanzas sobre posibles cambios.',
    url: 'https://www.bbc.com/mundo/noticias-internacional-59278657',
    urlToImage:
      'https://ichef.bbci.co.uk/news/1024/branded_mundo/289A/production/_121549301_gettyimages-1236516571.jpg',
    publishedAt: '2021-11-14T06:54:38Z',
    content:
      '<ul><li>Matt McGrath </li><li>Corresponsal de Medio Ambiente BBC</li></ul>Fuente de la imagen, Reuters Pie de foto, Manifestación cerca de la conferencia de la COP26 en apoyo de las víctimas de la… [+5456 chars]',
  },
  {
    author: 'null',
    title:
      'Músicos cubanos regalan concierto sinfónico a la capital de Cuba - Prensa Latina',
    description:
      'La Habana, 14 nov (Prensa Latina) Más de 100 músicos cubanos, bajo el liderazgo del maestro José María Vitier unirán su talento hoy para regalar un concierto sinfónico a la Villa de San cristóbal de La Habana por sus 502 años.',
    url: 'https://www.prensa-latina.cu/2021/11/14/musicos-cubanos-regalan-concierto-sinfonico-a-la-capital-de-cuba',
    urlToImage:
      'https://www.prensa-latina.cu/wp-content/uploads/2021/11/Cuba-Habana-Concerto-Jose-Maria-Vitier.jpg',
    publishedAt: '2021-11-14T05:12:06Z',
    content: null,
  },
  {
    author: 'null',
    title:
      'BioCubaFarma y su apuesta por la soberanía tecnológica - CubaDebate',
    description:
      'La Industria Biotecnológica y Farmacéutica es un sector de alta tecnología, su desarrollo y crecimiento sostenible requiere de la introducción sistemática de los más importantes avances tecnológicos. Para BioCubaFarma eso es una prioridad. No se puede aspirar…',
    url: 'http://www.cubadebate.cu/especiales/2021/11/14/biocubafarma-y-su-apuesta-por-la-soberania-tecnologica/',
    urlToImage:
      'http://www.cubadebate.cu/wp-content/uploads/2021/11/soberana-02-produccion-9.jpg',
    publishedAt: '2021-11-14T05:05:12Z',
    content:
      'Foto: Alejandro Azcuy/ Cubadebate.Cuba es un país bloqueado por la potencia más poderosa del mundo y no sabemos cuánto tiempo más se mantendrá esta situación. Por esa razón tenemos que lograr cada … [+8853 chars]',
  },
] as Definitions.Article[];
