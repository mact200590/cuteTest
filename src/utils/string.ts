// eslint-disable-next-line prettier/prettier
const STRINGS = {
  generals: {
    LINK_EXTERNAL: 'Ver el ariculo completo en la fuente >>',
  },
};

export default STRINGS;
