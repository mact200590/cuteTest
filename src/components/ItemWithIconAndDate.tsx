import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Text} from 'react-native-paper';
import moment from 'moment';
import IconButton from './atoms/IconButton';
import {IconNames} from '../utils/types';

const styles = StyleSheet.create({
  iconDateContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 5,
    alignItems: 'center',
  },
  date: {
    fontSize: 20,
    color: '#A0A0A0',
  },
});

interface Props {
  date: string;
  icon: IconNames;
  iconColor?: string;
  handleOnPress: () => void;
  id?: string;
}
const ItemWithIconAndDate = ({date, icon, iconColor, handleOnPress}: Props) => {
  return (
    <View style={styles.iconDateContainer}>
      <Text style={styles.date}>{moment(date).format('DD/MM/YYYY')}</Text>
      <IconButton
        width={40}
        height={40}
        icon={icon}
        onPress={handleOnPress}
        iconColor={iconColor}
      />
    </View>
  );
};

export default ItemWithIconAndDate;
