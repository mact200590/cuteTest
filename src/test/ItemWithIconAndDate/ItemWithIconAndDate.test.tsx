/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import renderer from 'react-test-renderer';
import ItemWithIconAndDate from '../../components/ItemWithIconAndDate';
import '@testing-library/react-native';
import {IconNames} from '../../utils/types';

describe('Test in <ItemWithIconAndDate/> component', () => {
  const handleOnPressMock = jest.fn();
  const dataMock = {
    date: '2019-05-14',
    icon: 'start',
  };
  test('To toMatchSnapshot', () => {
    const tree = renderer
      .create(
        <ItemWithIconAndDate
          date={dataMock.date}
          icon={dataMock.icon as IconNames}
          handleOnPress={handleOnPressMock}
          iconColor="red"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
