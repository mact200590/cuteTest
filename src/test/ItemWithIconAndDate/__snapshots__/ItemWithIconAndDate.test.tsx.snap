// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ItemWithIconAndDate/> component To toMatchSnapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 5,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Object {
          "color": "#A0A0A0",
          "fontSize": 20,
        },
      ]
    }
  >
    14/05/2019
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "minHeight": 20,
            "minWidth": 20,
          },
          undefined,
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={20}
        bbWidth={20}
        focusable={false}
        height={20}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 20,
              "width": 20,
            },
          ]
        }
        vbHeight={53.867}
        vbWidth={53.867}
        width={20}
        x="0px"
        y="0px"
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M26.934 1.318L35.256 18.182 53.867 20.887 40.4 34.013 43.579 52.549 26.934 43.798 10.288 52.549 13.467 34.013 0 20.887 18.611 18.182z"
            fill={4294901760}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
